src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -108,8 +150,17 @@ var DOMRenderer = ReactFiberReconciler({
     domElement : Instance,
     type : string,
     oldProps : Props,
-    newProps : Props
+    newProps : Props,
+    hostContext : string
   ) : boolean {
+    if (__DEV__) {
+      if (oldProps.children !== newProps.children && (
```

Comment:
I should probably check that we're moving *to* having a non-reified child here instead of always running the check.